src/grab.ts

```diff
@@ -1,157 +1,121 @@
 import { z } from "zod";
-import { BaseResult, ValueOf } from "./types";
+import { ValueOf } from "./types";
+import { ArrayResult, BaseResult, EntityResult } from "./builder";
```

Comment:
Regarding the class names, `*Result` was a bit confusing to me because it seems like it would be the actual output stored inside, rather than just the query and anticipated schema. Would you be open to alternate naming? The best I could come up with was `*Intent` or `*Plan`, though those may just trade in the misleading aspect for plain old confusion ("what's an `EntityIntent`?").

Perhaps there is precedent in other query builders that we could reference? I imagine many might call it `*Query`, which wouldn't be too bad either. Just a smidge confusing in that that class is more than just the query string.

Reply:
100%, I totally agree. I had that thought pop in my head at some point, but it must have gotten lost in the thick of things. Thanks for bringing it back up – I think I like `*Query`, that seems to make sense to me!